dev-docs/tls.md

```diff
@@ -0,0 +1,152 @@
+# TLS
+
+TLS stands for Transport Layer Security. We use TLS to encrypt all traffic
+between all services in our kubernetes (k8s) cluster.
+
+All traffic in k8s is intended to be encrypted. [PR
+8561](https://github.com/hail-is/hail/pull/8561) started this work and has a
+thorough explanation of the architecture at that time. Currently the only
+service that intentionally does not accept incoming TLS traffic is the internal
+gateway. Moreover, batch workers, which exist on GCE VMs rather than inside k8s,
+do not encrypt traffic to the internal gateway. The internal gateway *does*
+encrypt traffic as it is forwarded to a k8s-resident service.
+
+## Create a Self-Signed x509 Certificate in PEM Format
+
+Produce an x509 certificate. The key is a 4096-bit RSA key. The cert is valid
+from today until 365 days from today. The server must be accessed via the domain
+name `localhost`. A client that verifies hostnames will reject this certificate
+if the server is accessed via an IP (like `127.0.0.1`) or other names (like
+`wm06b-953`). The certificate is not password protected due to `-nodes`.
+
+```
+openssl req -x509 \
+        -newkey rsa:4096 \
+        -keyout server-key.pem \
+        -out server-cert.pem \
+        -days 365 \
+        -subj '/CN=localhost' \
+        -nodes
+```
+
+## Bundle a Key and Certificate into a PKCS12 File
+
+Create a PKCS12 file. PKCS12 files are primarily useful for creating instances
+of a Java `KeyStore`. It is not possible to [elide a
+password](https://stackoverflow.com/questions/27497723/export-a-pkcs12-file-without-an-export-password)
+empty passwords are not recommended.
```

Comment:
You're missing some punctuation?  "It is not possible to elide a password empty passwords are not recommended."